fsm.go

```diff
@@ -127,17 +129,44 @@ func (r *fsmS) announceSensor(e *f.Event) {
 	log.debug("announcing sensor to the agent")
 
 	go func(cb func(b bool, from *FromS)) {
+
 		d := &Discovery{
 			PID:  os.Getpid(),
 			Name: os.Args[0],
 			Args: os.Args[1:]}
 
+		var socket net.Conn
```

Comment:
We're opening an additional dummy socket here before the HTTP announce is made.  We pass the fd/inode of this dummy socket in the announce and keep it open until the announce response is received.  

This is far simpler than modifying the announce payload before it's transmitted (but after socket is open).

So the cost is one extra socket connection on announce but I think it's fair given announce is only every 30 seconds or so.